activesupport/lib/active_support/cache.rb

```diff
@@ -249,27 +253,79 @@ def retrieve_pool_options(options)
+      #   and +inflate+.
+      #
+      #   The default compressor is +Zlib+. To define a new custom compressor
+      #   that also decompresses old cache entries, you can check compressed
+      #   values for Zlib's <tt>"\x78"</tt> signature:
+      #
+      #     module MyCompressor
+      #       def self.deflate(dumped)
+      #         # compression logic... (make sure result does not start with "\x78"!)
+      #       end
+      #
+      #       def self.inflate(compressed)
+      #         if compressed.start_with?("\x78")
+      #           Zlib.inflate(compressed)
+      #         else
+      #           # decompression logic...
+      #         end
+      #       end
+      #     end
+      #
+      #     ActiveSupport::Cache.lookup_store(:redis_cache_store, compressor: MyCompressor)
+      #
       # [+:coder+]
-      #   Replaces the default serializer for cache entries. +coder+ must
-      #   respond to +dump+ and +load+. Using a custom coder disables automatic
-      #   compression.
+      #   The coder for serializing and (optionally) compressing cache entries.
+      #   Must respond to +dump+ and +load+.
+      #
+      #   The default coder composes the serializer and compressor, and includes
+      #   some performance optimizations. If you only need to override the
+      #   serializer or compressor, you should specify the +:serializer+ or
+      #   +:compressor+ options instead.
       #
-      #   Alternatively, you can specify <tt>coder: :message_pack</tt> to use a
-      #   preconfigured coder based on ActiveSupport::MessagePack that supports
-      #   automatic compression and includes a fallback mechanism to load old
-      #   cache entries from the default coder. However, this option requires
-      #   the +msgpack+ gem.
+      #   The +:coder+ option is mutally exclusive with the +:serializer+ and
+      #   +:compressor+ options. Specifying them together will raise an
+      #   +ArgumentError+.
       #
       # Any other specified options are treated as default options for the
       # relevant cache operations, such as #read, #write, and #fetch.
       def initialize(options = nil)
-        @options = options ? normalize_options(options) : {}
+        @options = options ? validate_options(normalize_options(options)) : {}
 
         @options[:compress] = true unless @options.key?(:compress)
         @options[:compress_threshold] ||= DEFAULT_COMPRESS_LIMIT
 
-        @coder = @options.delete(:coder) { default_coder } || :passthrough
-        @coder = Cache::SerializerWithFallback[@coder] if @coder.is_a?(Symbol)
+        @coder = @options.delete(:coder) do
+          legacy_serializer = Cache.format_version < 7.1 && !@options[:serializer]
+          serializer = @options.delete(:serializer) || default_serializer
+          serializer = Cache::SerializerWithFallback[serializer] if serializer.is_a?(Symbol)
+          compressor = @options.delete(:compressor) || Zlib
```

Comment:
```suggestion
          compressor = @options.delete(:compressor) { Zlib }
```

I think we want to allow `compressor: nil` as an option.

Reply:
I've made this change, but is there a benefit to specifying `compressor: nil` instead of `compress: false`?  Are you thinking it may be easier for downstream code to specify `compressor: nil`?

Reply:
Yeah, I think `compress: false` becomes redundant with `compressor: nil`, but not worth deprecating I think.